Taskfile.yml

```diff
@@ -324,56 +322,50 @@ tasks:
       - pre-commit run -a
 
   run-website:
-    desc: Build & serve the static website.
+    desc: Build & serve the static website for interactive development.
     dir: web/website
     cmds:
       - hugo server
 
-  run-web:
-    desc: Build & serve the website & playground.
-    summary:
-      Note that this only live-reloads the static website; and requires
-      rerunning to pick up playground & book changes.
-    dir: web/website
-    cmds:
-      - task: build-web
-      # Then start web server with rendering to disk, so it picks up the playground files.
-      - hugo server --renderToDisk
-
   run-book:
-    desc: Build & serve the book.
+    desc: Build & serve the book for interactive development.
     dir: web/book
     cmds:
       - mdbook serve --port=3000
 
   run-playground:
-    desc: Build & serve the playground.
-    dir: web/playground
-    cmds:
-      - npm install
-      - npm start
-
-  run-playground-cached:
-    desc: Build & serve the playground, without rebuilding rust code.
+    desc: Build & serve the playground for interactive development.
     dir: web/playground
     cmds:
-      - task: install-playground-npm-dependencies
+      - task update-playground-dependencies
       - npm start
 
-  install-playground-npm-dependencies:
-    # Look to see if npm dependencies need to be updated
-    # Use task's sources/generates to see if checksums of
-    # node_modules directory have changed since package.json was updated
-
-    desc: Check to see if package.json has changed since we ran npm install
+  update-playground-dependencies:
+    # Check if npm dependencies for the playground need to be updated
+    # Use task's sources/generates to see if package.json,
+    # or anything in crates or bindings was updated after the
+    # node_modules was rebuilt
+    desc: Install dep's if something important changed recently
     dir: web/playground
     cmds:
-      - npm install
+      - task install-playground-dependencies-all
     sources:
       - package.json
+      - ../../crates/**/*.rs
+      - ../../bindings/**/*.rs
     generates:
       - node_modules/*
 
+  install-playground-dependencies-all:
```

Comment:
Can we just combine this with the previous? Passing `--force` can override the `sources` check if we ever want to force an install

Reply:
I think I'd like to leave it like this. I factored it out when `--install-links=false` was necessary (only wanted to have that magic invocation in one place).

Reply:
What's the difference between calling one vs the other?

Reply:
I don't understand the thrust of this question: "combine with the previous"? Which two are you suggesting could be combined?

Here's my understanding of the situation: In the original _Taskfile.yaml_ there were several invocations of `npm install --install-links=false` which I factored into a single `install-playground-dependencies-all` task so the magic `--install-links...` wouldn't be screwed up in one of the multiple places. I also described that option in the comments (again, in one place). Additionally, I changed all the places that called `npm install...` to use the `install-playground-dependencies` task so they could take advantage of caching prior results.

Now that the `--install-links=false` is no longer required, we certainly could get rid of the (single) call to `install-playground-dependencies-all` and replace it with a simple `npm install`

But I'll also point out the "documentation advantage" of having a named task that unconditionally runs the proper `npm install` command in the case that the logic behind the `install-playground-dependences` gets screwed up.  Instead of needing to remember where the `--force` option might be used, people can just `task install-playground-dependencies-all` to un-screw it. 

I hope that addresses your question

Reply:
What's the difference between `install-playground-dependencies-all` and `update-playground-dependencies`?

Reply:
Can we combine these though? I don't see why we have a task that seems to just wrap another task?